src/query/plan/rewrite/index_lookup.hpp

```diff
@@ -584,44 +607,74 @@ class IndexLookupRewriter final : public HierarchicalLogicalOperatorVisitor {
           // cannot scan `n` by property index.
           continue;
         }
+
         const auto &property = filter.property_filter->property_;
         if (!db_->LabelPropertyIndexExists(GetLabel(label), GetProperty(property))) {
           continue;
         }
-        auto is_better_type = [&found](PropertyFilter::Type type) {
-          // Order the types by the most preferred index lookup type.
-          static const PropertyFilter::Type kFilterTypeOrder[] = {
-              PropertyFilter::Type::EQUAL, PropertyFilter::Type::RANGE, PropertyFilter::Type::REGEX_MATCH};
-          auto *found_sort_ix = std::find(kFilterTypeOrder, kFilterTypeOrder + 3, found->filter.property_filter->type_);
-          auto *type_sort_ix = std::find(kFilterTypeOrder, kFilterTypeOrder + 3, type);
-          return type_sort_ix < found_sort_ix;
-        };
-
-        int64_t vertex_count = db_->VerticesCount(GetLabel(label), GetProperty(property));
-        std::optional<storage::LabelPropertyIndexStats> new_stats =
-            db_->GetIndexStats(GetLabel(label), GetProperty(property));
-
-        // Conditions, from more to less important:
-        // the index with 10x less vertices is better.
-        // the index with smaller average group size is better.
-        // the index with equal avg group size and distribution closer to the uniform is better.
-        // the index with less vertices is better.
-        // the index with same number of vertices but more optimized filter is better.
-        if (!found || vertex_count * 10 < found->vertex_count) {
-          found = LabelPropertyIndex{label, filter, vertex_count, new_stats};
-          continue;
-        }
 
-        if (int cmp_res = compare_indices(found, new_stats, vertex_count);
-            cmp_res == -1 ||
-            cmp_res == 0 && (found->vertex_count > vertex_count ||
-                             found->vertex_count == vertex_count && is_better_type(filter.property_filter->type_))) {
-          found = LabelPropertyIndex{label, filter, vertex_count, new_stats};
-        }
+        candidate_indices.emplace_back(std::make_pair(
+            IndexHint{.index_type_ = IndexHint::IndexType::LABEL_PROPERTY, .label_ = label, .property_ = property},
+            filter));
+        candidate_index_lookup.insert({std::make_pair(label, property), filter});
+      }
+    }
+
+    for (const auto &[index_type, label, maybe_property] : index_hints_) {
+      if (index_type != IndexHint::IndexType::LABEL_PROPERTY) continue;
+      auto property = *maybe_property;
+
+      if (!db_->LabelPropertyIndexExists(GetLabel(label), GetProperty(property))) {
+        throw QueryRuntimeException("Index for label {} and property {} doesn't exist", label.name, property.name);
```

Comment:
to aggresive, I would rather log it with spdlog::debug and proceed next to see what happens with the plan

Reply:
it's just a hint, it's not a query breaker

Reply:
Changed